File: client_sdk/java/the-client/src/vip/dummy/texasholdem/playerai/PlayerAI.java
package vip.dummy.texasholdem.playerai;

import vip.dummy.texasholdem.IndicationCallbacks;
import vip.dummy.texasholdem.indication.*;
import vip.dummy.texasholdem.message.ActionMessage;

public class PlayerAI implements IndicationCallbacks {

    private static PlayerAI playerAI;

    public static PlayerAI getInstance() {
        if (null == playerAI) {
            playerAI = new PlayerAI();
        }
        return playerAI;
    }

    public ActionMessage nextStep(String serverMessage) {
        return null;
    }

    @Override
    public void onNewPeer(NewPeerIndication newPeerIndication) {

    }

    @Override
    public void onNewRound(NewRoundIndication newRoundIndication) {

    }

    @Override
    public void onStartReload(StartReloadIndication startReloadIndication) {

    }

    @Override
    public void onDeal(DealIndication dealIndication) {

    }

    @Override
    public void onAction(ActionIndication actionIndication) {

    }

    @Override
    public void onBet(BetIndication betIndication) {

    }

    @Override
    public void onShowAction(ShowActionIndication showActionIndication) {

    }

    @Override
    public void onRoundEnd(RoundEndIndication roundEndIndication) {

    }

    @Override
    public void onGameOver(GameOverIndication gameOverIndication) {

    }
}

File: client_sdk/python/the-client/src/vip/dummy/texasholdem/playerai/PlayerAI.py
import sys

sys.path.append('../')
sys.path.append('../message')
sys.path.append('../message/data')
from IndicationCallbacks import IndicationCallbacks
from ReloadMessage import ReloadMessage
from ReloadData import ReloadData
from ActionMessage import ActionMessage
from ActionData import ActionData


class PlayerAI(IndicationCallbacks):
    def __init__(self, webSocketClient):
        self.webSocketClient = webSocketClient

    def nextStep(self):
        self.allIn()

    def Reload(self):
        reloadMessage = ReloadMessage(ReloadData())
        self.webSocketClient.send(reloadMessage.toJson())

    def Call(self):
        actionMessage = ActionMessage(ActionData("call", 0))
        self.webSocketClient.send(actionMessage.toJson())

    def Raise(self):
        actionMessage = ActionMessage(ActionData("raise", 0))
        self.webSocketClient.send(actionMessage.toJson())

    def Fold(self):
        actionMessage = ActionMessage(ActionData("fold", 0))
        self.webSocketClient.send(actionMessage.toJson())

    def Check(self):
        actionMessage = ActionMessage(ActionData("check", 0))
        self.webSocketClient.send(actionMessage.toJson())

    def Bet(self, amount):
        actionMessage = ActionMessage(ActionData("bet", amount))
        self.webSocketClient.send(actionMessage.toJson())

    def AllIn(self):
        actionMessage = ActionMessage(ActionData("allin", 0))
        self.webSocketClient.send(actionMessage.toJson())

    def onNewPeer(self, newPeerIndication):
        print("new peer")

    def onNewRound(self, newRoundIndication):
        print("<< on new round")

    def onStartReload(self, startReloadIndication):
        print("<< on reload indication")

    def onDeal(self, dealIndication):
        print("<< on deal")

    def onAction(self, actionIndication):
        print("<< on action")

    def onBet(self, betIndication):
        print("<< on bet")

    def onShowAction(self, showActionIndication):
        print("<< on show action")

    def onRoundEnd(self, roundEndIndication):
        print("<< on round end")

    def onGameOver(self, gameOverIndication):
        print("<< on game over")


if __name__ == "__main__":
    test = None
    if test.strip() == '':
        print "test"
